st2common/st2common/services/keyvalues.py

```diff
@@ -42,15 +46,72 @@ def _get(self, name):
         # the lookup is for 'key_base.key_value' it is likely that the calling code, e.g. Jinja,
         # will expect to do a dictionary style lookup for key_base and key_value as subsequent
         # calls. Saving the value in cache avoids extra DB calls.
-        return KeyValueLookup(key, self._value_cache)
+        return KeyValueLookup(key_prefix=key, cache=self._value_cache, scope=self._scope)
 
     def _get_kv(self, key):
-        kvp = None
-        try:
-            kvp = KeyValuePair.get_by_name(key)
-        except ValueError:
-            # ValueErrors are expected in case of partial lookups
-            pass
-        # A good default value for un-matched value is empty string since that will be used
-        # for rendering templates.
+        scope = self._scope
+        kvp = KeyValuePair.get_by_scope_and_name(scope=scope, name=key)
         return kvp.value if kvp else ''
+
+
+class UserKeyValueLookup(object):
```

Comment:
I would personally move those look-up classes somewhere outside of `services` module since it's more like Jinja specific functionality...